libdrgn/arch_s390_defs.py

```diff
@@ -0,0 +1,16 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+REGISTERS = [
+    *[DrgnRegister(f"r{i}") for i in range(0,16)],
+    DrgnRegister("pswm"),
+    DrgnRegister("pswa"),
+    DrgnRegister("bc"),
+]
+
+REGISTER_LAYOUT = [
+    # Caller-saved registers.
+    *[DrgnRegisterLayout(f"r{i}", size=8, dwarf_number=i) for i in range(0,16)],
+    DrgnRegisterLayout("pswm", size=8, dwarf_number=64),
+    DrgnRegisterLayout("pswa", size=8, dwarf_number=65),
+    DrgnRegisterLayout("bc", size=8, dwarf_number=127),
```

Comment:
I don't see any mentions of DWARF register number 127, so I suspect you did this because it's a "fake" register. You can use `dwarf_number=None` to indicate that it's not used by DWARF.

However, I'm not sure whether we need this at all. Can't we get the backchain to the next frame by reading from `r15 + 152` at the current frame?